### PR TITLE
✨ Add controller tests for HelmChartProxy

### PR DESCRIPTION
**Pull request by Jont828**
**What this PR does / why we need it**: Add controller tests for `helmchartproxy_controller.go` and `helmchartproxy_controller_phases.go`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by Jont828** [comment]
@CecileRobertMichon Could I get a review on this? Not urgent so no rush

File: controllers/helmchartproxy/helmchartproxy_controller.go [hunk 1]
```diff
@@ -279,6 +279,7 @@ func (r *HelmChartProxyReconciler) listInstalledReleases(ctx context.Context, na
 func (r *HelmChartProxyReconciler) aggregateHelmReleaseProxyReadyCondition(ctx context.Context, helmChartProxy *addonsv1alpha1.HelmChartProxy) error {
 	log := ctrl.LoggerFrom(ctx)
 
+	fmt.Printf("Aggregating HelmReleaseProxyReadyCondition\n")
```
Comment (CecileRobertMichon) [inline controllers/helmchartproxy/helmchartproxy_controller.go:282]: why the `fmt.Printf` lines here? logs should be enough

File: controllers/helmchartproxy/helmchartproxy_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+		{
+			name:           "successfully delete orphaned HelmReleaseProxies",
+			helmChartProxy: defaultProxy,
+			objects:        []client.Object{hrpReady1, hrpReady2},
+			expect: func(g *WithT, c client.Client, hcp *addonsv1alpha1.HelmChartProxy) {
+				g.Expect(hcp.Status.MatchingClusters).To(BeEmpty())
+				g.Expect(c.Get(ctx, client.ObjectKey{Namespace: hrpReady1.Namespace, Name: hrpReady1.Name}, &addonsv1alpha1.HelmReleaseProxy{})).ToNot(Succeed())
+				g.Expect(c.Get(ctx, client.ObjectKey{Namespace: hrpReady2.Namespace, Name: hrpReady2.Name}, &addonsv1alpha1.HelmReleaseProxy{})).ToNot(Succeed())
+
+				// Vacuously true as there are no HRPs
```
Comment (CecileRobertMichon) [inline controllers/helmchartproxy/helmchartproxy_controller_test.go:300]: > Vacuously

learned a new word today :)

File: controllers/helmchartproxy/helmchartproxy_controller.go (hunk 1 above)
Comment (Jont828) [reply controllers/helmchartproxy/helmchartproxy_controller.go]: Forgot to remove debug logs! Side question -- is there a way to get V(2) logs to show up when running unit tests/make test? IIRC I added the print statements to ensure that the messages show up.

Comment (CecileRobertMichon) [reply controllers/helmchartproxy/helmchartproxy_controller.go]: I don't think so... at least not that I know of

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Pull request merged**